examples/pong_tutorial_03/systems/paddle.rs

```diff
@@ -1,36 +1,30 @@
 use crate::pong::{Paddle, Side, ARENA_HEIGHT, PADDLE_HEIGHT};
 use amethyst::{
     core::transform::Transform,
-    derive::SystemDesc,
-    ecs::prelude::{Join, Read, ReadStorage, System, SystemData, WriteStorage},
-    input::{InputHandler, StringBindings},
+    ecs::{Runnable, SystemBuilder},
+    input::{get_input_axis_simple, InputHandler, StringBindings},
+    prelude::*,
 };
 
-#[derive(SystemDesc)]
-pub struct PaddleSystem;
-
-impl<'s> System<'s> for PaddleSystem {
-    type SystemData = (
-        WriteStorage<'s, Transform>,
-        ReadStorage<'s, Paddle>,
-        Read<'s, InputHandler<StringBindings>>,
-    );
-
-    fn run(&mut self, (mut transforms, paddles, input): Self::SystemData) {
-        for (paddle, transform) in (&paddles, &mut transforms).join() {
-            let movement = match paddle.side {
-                Side::Left => input.axis_value("left_paddle"),
-                Side::Right => input.axis_value("right_paddle"),
-            };
-            if let Some(mv_amount) = movement {
-                let scaled_amount = 1.2 * mv_amount;
+pub fn build() -> impl Runnable {
+    SystemBuilder::new("PaddleSystem")
+        .with_query(<(&Paddle, &mut Transform)>::query())
+        .read_resource::<InputHandler<StringBindings>>()
+        .read_component::<Paddle>()
+        .write_component::<Transform>()
+        .build(move |_commands, world, input, query_paddles| {
```

Comment:
When I was reading this blog post 
https://amethyst.rs/posts/legion-ecs-v0.3
I thought that in Legion systems are just simple Rust functions with attached macros and specific arguments. 

Is SystemBuilder a different way to do the same thing?
And if so, why did you choose this other solution?